runtime/src/bank/accounts_lt_hash.rs

```diff
@@ -313,17 +316,11 @@ impl Bank {
 
         // Only insert the account the *first* time we see it.
         // We want to capture the value of the account *before* any modifications during this slot.
-        let (is_in_cache, lookup_time) = meas_dur!({
-            self.cache_for_accounts_lt_hash
-                .read()
-                .unwrap()
-                .contains_key(address)
-        });
+        let (is_in_cache, lookup_time) =
+            meas_dur!(self.cache_for_accounts_lt_hash.contains_key(address));
```

Comment:
just to validate my understanding of this code:

contains_key() else entry().or_insert_with() is racy (even prior to this PR), in that one thread could see contains_key()=false, but then another thread might be faster to get to entry().or_insert_with().

Is the assumption that in the caller (tx processing), accounts are write-locked, so the race can only happen with readonly accounts, which are guaranteed to have identical `AccountState`? If that's the case, I'd add a comment that explains this.